data_structures/src/transaction_factory.rs

```diff
@@ -247,7 +261,27 @@ pub fn get_total_balance(all_utxos: &UnspentOutputsPool, pkh: PublicKeyHash) ->
                 None
             }
         })
-        .sum()
+        .sum();
+
+    // Get the balance of the confirmed utxo set from storage
+    let old_balance = match old_all_utxos {
+        Some(old_all_utxos) => old_all_utxos
```

Comment:
I tend to use If let Some(old_all_utxos) = old_all_utxos for Options but its just a matter of feeling (I would rather use match for more than 2 patterns).  But it is a matter of style